src/webapi/Controllers/AuthorizationController.cs

```diff
@@ -39,36 +41,77 @@ public class AuthorizationController : Controller
     [HttpPost($"~{TokenPath}"), IgnoreAntiforgeryToken, Produces("application/json")]
     public async Task<IActionResult> Exchange([FromForm] PasswordTokenRequest tokenRequest)
     {
-        if (tokenRequest.GrantType != GrantTypes.Password)
+        var openIdRequest = HttpContext.GetOpenIddictServerRequest();
+        if (openIdRequest?.GrantType != GrantTypes.Password && openIdRequest?.GrantType != GrantTypes.RefreshToken)
         {
             return BadRequest(new { error = "The application only accepts the password grant type at this time." });
         }
 
         var user = await _userManager.FindByNameAsync(tokenRequest.Username);
```

Comment:
if this is a refresh token call, we immediately null the user out. Can we move this call under the `else if (openIdRequest.GrantType == GrantTypes.Password)` call on line 88

Reply:
Done!